akka-actor-typed/src/main/scala/akka/actor/typed/MessageAndSignals.scala

```diff
@@ -71,20 +80,26 @@ case object PostStop extends PostStop {
  *
  * @param ref Scala API: the `ActorRef` for the terminated actor
  */
-final case class Terminated(ref: ActorRef[Nothing])(failed: Throwable) extends Signal {
-  /**
-   * Scala API: If the watched actor is a direct child, and was stopped because it failed, this will contain the
-   * Exception it failed with, for all other cases it will be `None`.
-   */
-  def failure: Option[Throwable] = Option(failed)
-
+@DoNotInherit
+sealed class Terminated(val ref: ActorRef[Nothing]) extends Signal {
```

Comment:
How about make it a `sealed trait` instead of class?

Reply:
I think it's better to introduce the whole `Terminated` tree as a kind of `Signal`, and split with `ChildTerminated` and `LocalTerminated`

Reply:
There's remove termination to and i don't think we should distinguish between local and remote termination as it breaks location transparency.

Reply:
OK:), Currently is perfect too:)

Reply:
Remote*